### PR TITLE
feat(core): add a new package

### PR DESCRIPTION
**Pull request by dsseng**

### CODE REVIEW

**Comment by ElijahKotyluk** [comment]
Will complete conversion when I wake up and finish rewriting the new code in TS.

**Review by ElijahKotyluk — COMMENTED** [review]
@sh7dm

File: packages/core/src/runnable.ts [hunk 1]
```diff
@@ -23,12 +27,23 @@ export default class Runnable {
    * @param {Function} fn
    * @return {Result}
    */
-  public run(fn: any): Result {
-    // run runnable instance
+  public run(fn: () => any | any[]): Result {
```
Comment (dsseng) [inline packages/core/src/runnable.ts]: just `any`, it can mean array too

File: packages/core/src/runnable.ts [hunk 2]
```diff
@@ -23,12 +27,23 @@ export default class Runnable {
...
     try {
       fn();
     } catch (error) {
       return new Result('Failed', error);
     }
+
+    return new Result('Success', [`passing test`]);
```
Comment (dsseng) [inline packages/core/src/runnable.ts]: We need an enum here, no strings, TS rules

File: packages/core/src/runnable.ts [hunk 3]
```diff
@@ -23,12 +27,23 @@ export default class Runnable {
...
+  }
+
+  /**
+   * Run asynchronous `Test` or `Suite`:
+   * @public
+   * @param {Function} fn 
+   * @return {Result}
+   */
+  public async asyncRun(fn: any[]): Promise<Result> {
```
Comment (dsseng) [inline packages/core/src/runnable.ts]: why array? it should be `fn: () => any`

File: packages/core/src/runnable.ts [hunk 4]
```diff
@@ -8,13 +8,17 @@ import Result from './result';
 
 export default class Runnable {
   public description: string;
-  public fn: () => Result | Result;
+  public fn: () => any;
   public pending: boolean;
+  public async: boolean | number;
+  public sync: boolean;
```
Comment (dsseng) [inline packages/core/src/runnable.ts]: no need for these ones

File: packages/core/test/suite.spec.ts [hunk 5]
```diff
@@ -3,30 +3,59 @@ import Runnable from '../src/runnable';
 import Suite from '../src/suite';
 
 describe('Suite', () => {
-  it('to be an instance of Runnable', () => {
+  it('should be an instance of Runnable', () => {
```
Comment (dsseng) [inline packages/core/test/suite.spec.ts]: imho no need to test for it

File: packages/core/test/suite.spec.ts [hunk 6]
```diff
@@ -3,30 +3,59 @@ import Runnable from '../src/runnable';
...
     expect(suite.run()).toBeInstanceOf(Result);
+    expect(suite.asyncRun()).toBeInstanceOf(Promise);
+    expect(suite.skip).toBeFalsy();
+    expect(suite).toHaveProperty('description');
+    expect(suite).toHaveProperty('fn');
+    expect(suite).toHaveProperty('result');
   });
-  it('to have the type set to `Suite`', () => {
-    const suite = new Suite('Suite', () => null, false);
+  it('should be type `Suite`', () => {
```
Comment (dsseng) [inline packages/core/test/suite.spec.ts]: and this

File: packages/core/test/suite.spec.ts [hunk 7]
```diff
@@ -3,30 +3,59 @@ import Runnable from '../src/runnable';
...
+    const suite = new Suite('Suite', () => null);
 
     expect(suite).toHaveProperty('type');
     expect(suite.type).toBe('Suite');
   });
-  it('to have a description', () => {
-    const suite = new Suite('Suite', () => null, false);
+  it('should skip', () => {
```
Comment (dsseng) [inline packages/core/test/suite.spec.ts]: inherited. tested in Runnable

File: packages/core/test/suite.spec.ts [hunk 8]
```diff
@@ -3,30 +3,59 @@ import Runnable from '../src/runnable';
...
+    const fn = jest.fn();
+    const suite = new Suite('Suite', fn);
 
-    expect(suite).toHaveProperty('description');
-    expect(suite.description).toBe('Suite');
+    expect(suite.skip = true).toBeTruthy();
+    expect(suite.run()).toEqual(new Result(Status.Skipped, []));
+  });
+  it('should error', () => {
```
Comment (dsseng) [inline packages/core/test/suite.spec.ts]: the same, tested before

**Review by ElijahKotyluk — APPROVED** [review]

**Pull request merged**